File: front-end/src/app/models/Candidat.ts
import { Role } from "./Role";
import { User } from "./User";

export class Candidat extends User {
    public nomPrenom : string ;
    public  numCIN : string ;
	public dateNaissance:  Date  ;
	public  nomPere : string ;
	public prenomPere : string ;
	public  numTel : string ;
	public  adress : string ;
	
    constructor(
        numCIN : string,
        nomPrenom : string,
        nomPere : string ,
        prenomPere : string ,
        numTel : string ,
        adress : string ,
        dateNaissance : Date,
        id? : number,
        username? : string,
        email? : string,
        password? : string,
        role? : string[],
        
    ){
        super(id,username,email,password,role);
        
        this.nomPrenom = nomPrenom;
        this.numCIN = numCIN;
        this.dateNaissance = dateNaissance;
        this.nomPere = nomPere;
        this.prenomPere = prenomPere;
        this.numTel = numTel;
        this.adress = adress;
    }
}

File: back-end/src/main/java/org/example/entities/Candidat.java
package org.example.entities;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

@Data @NoArgsConstructor @AllArgsConstructor
@Entity
@Table(name = "candidats", uniqueConstraints = {
        @UniqueConstraint(columnNames = "numCIN")
})

@Inheritance(strategy = InheritanceType.JOINED)
public class Candidat extends User{

    @Column
    private String numCIN;

    @Column
    private LocalDateTime dateNaissance;

    @Column
    private String nomPrenom;

    @Column
    private String nomPere;

    @Column
    private String prenomPere;

    @Column
    private String numTel;

    @Column
    private String adress;
}
